### PR TITLE
ESLINTJS-53 Fix incompatibilities with ESLint 9

### PR DESCRIPTION
**Pull request by vdiez**
Fixes https://sonarsource.atlassian.net/browse/ESLINTJS-53
Fixes https://sonarsource.atlassian.net/browse/JS-297

### CODE REVIEW

File: packages/jsts/src/rules/helpers/decorators/interceptor.ts [hunk 1]
```diff
@@ -93,6 +93,11 @@ export function interceptReport(
             return originalContext.sourceCode;
           },
 
+          // @ts-ignore
+          getSource(...args) {
+            return originalContext.sourceCode.getText(...args);
+          },
+
```
Comment (zglicz) [inline packages/jsts/src/rules/helpers/decorators/interceptor.ts:100]: above you are using `originalContext.getAncestors();`, `originalContext.getScope(); ` and `originalContext.markVariableAsUsed(name); `
Apparently, these now requires a node as an argument https://eslint.org/docs/latest/use/migrate-to-9.0.0#-removed-multiple-context-methods

**Review by zglicz — COMMENTED** [review]
Could you see if these are real issues that would fail in eslint v9

**Comment by vdiez** [comment]
Sorry there are many changes related completly removing ESLint ids, work started in [JS-278](https://github.com/SonarSource/SonarJS/pull/4769/files)

File: packages/jsts/src/rules/helpers/decorators/interceptor.ts (hunk 1 above)
Comment (vdiez) [reply packages/jsts/src/rules/helpers/decorators/interceptor.ts:100]: the ones in sourceCode require a node. This is overriding the context and forwarding same call. Tomorrow I'll check how many rules of the decorated rules use those methods.

File: packages/jsts/tests/linter/fixtures/wrapper/eslint.js
Comment (ericmorand-sonarsource) [inline packages/jsts/tests/linter/fixtures/wrapper/eslint.js:1]: Out of curiosity, were these fixtures not used?

**Pull request merged**